vendor/bat-native-ledger/src/bat/ledger/internal/bat_contribution_unittest.cc

```diff
@@ -172,4 +202,68 @@ TEST_F(BatContributionTest, WillTriggerNotification) {
       100, 4, 100, 5, {1, 5, 10, 20, 50}, 5, 90));
 }
 
+TEST_F(BatContributionTest, GetStatisticalVotingWinners) {
+  auto bat_contribution =
+      std::make_unique<braveledger_bat_contribution::BatContribution>(nullptr);
+
+  braveledger_bat_helper::PublisherList publisher_list;
+  PopulatePublisherList(&publisher_list);
+
+  braveledger_bat_helper::WINNERS_ST winner;
+  bool result = bat_contribution->GetStatisticalVotingWinner(
+      0.01, publisher_list, &winner);
+  EXPECT_TRUE(result);
+  EXPECT_STREQ(winner.publisher_data_.id_.c_str(), "publisher1");
+
+  result = bat_contribution->GetStatisticalVotingWinner(
+      0.05, publisher_list, &winner);
+  EXPECT_TRUE(result);
+  EXPECT_STREQ(winner.publisher_data_.id_.c_str(), "publisher2");
+
+  result = bat_contribution->GetStatisticalVotingWinner(
+      0.10, publisher_list, &winner);
+  EXPECT_TRUE(result);
+  EXPECT_STREQ(winner.publisher_data_.id_.c_str(), "publisher2");
+
+  result = bat_contribution->GetStatisticalVotingWinner(
+      0.20, publisher_list, &winner);
+  EXPECT_TRUE(result);
+  EXPECT_STREQ(winner.publisher_data_.id_.c_str(), "publisher3");
+
+  result = bat_contribution->GetStatisticalVotingWinner(
+      0.30, publisher_list, &winner);
+  EXPECT_TRUE(result);
+  EXPECT_STREQ(winner.publisher_data_.id_.c_str(), "publisher4");
+
+  result = bat_contribution->GetStatisticalVotingWinner(
+      0.40, publisher_list, &winner);
+  EXPECT_TRUE(result);
+  EXPECT_STREQ(winner.publisher_data_.id_.c_str(), "publisher4");
+
+  result = bat_contribution->GetStatisticalVotingWinner(
+      0.50, publisher_list, &winner);
+  EXPECT_TRUE(result);
+  EXPECT_STREQ(winner.publisher_data_.id_.c_str(), "publisher4");
+
+  result = bat_contribution->GetStatisticalVotingWinner(
+      0.60, publisher_list, &winner);
+  EXPECT_TRUE(result);
+  EXPECT_STREQ(winner.publisher_data_.id_.c_str(), "publisher5");
+
+  result = bat_contribution->GetStatisticalVotingWinner(
+      0.70, publisher_list, &winner);
+  EXPECT_TRUE(result);
+  EXPECT_STREQ(winner.publisher_data_.id_.c_str(), "publisher5");
+
+  result = bat_contribution->GetStatisticalVotingWinner(
+      0.80, publisher_list, &winner);
+  EXPECT_TRUE(result);
+  EXPECT_STREQ(winner.publisher_data_.id_.c_str(), "publisher5");
+
+  result = bat_contribution->GetStatisticalVotingWinner(
+      0.90, publisher_list, &winner);
+  EXPECT_TRUE(result);
+  EXPECT_STREQ(winner.publisher_data_.id_.c_str(), "publisher5");
+}
+
```

Comment:
This looks great.  One nit: Can you make the test a little conciser by using an array of cases?

```
struct {
  double dart;
  const char *publisher;
} cases[] = {
  { 0.01, "publisher1" },
  { 0.05, "publisher2" },
  { 0.10, "publisher3" },
  ...
};

for (size_t i = 0; i < arraycount(cases); i++) {
  ...
}
```

Reply:
Great idea, thanks.